sms_sponsorship/forms/sms_sponsorship_registration_form.py

```diff
@@ -171,7 +173,7 @@ def form_before_create_or_update(self, values, extra_values):
             partner_vals = self._get_partner_vals(extra_values)
             if partner:
                 # update existing partner
-                partner.write(partner_vals)
+                self.with_delay().update_partner(partner_vals)
```

Comment:
I think it would be best to put the update_partner method inside the res.partner object so that you can do `partner.with_delay().update_partner(partner_vals)`